Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/SqlServer/SqlDialectTest.cs

```diff
@@ -49,5 +57,219 @@ public void GetParameterName ()
+
+      commandMock.Verify();
+      dataParameterMock.Verify();
+
+      Assert.That(result, Is.SameAs(dataParameterMock.Object));
+      dataParameterMock.VerifySet(_ => _.Size = It.IsAny<int>(), Times.Never());
+    }
+
+    [Test]
+    public void CreateDataParameter_WithNullResult ()
+    {
+      var commandMock = new Mock<IDbCommand>(MockBehavior.Strict);
+      var dataParameterMock = new Mock<IDbDataParameter>(MockBehavior.Strict);
+
+      _typeConverterStub.Setup(_ => _.ConvertTo(null, null, "value", _storageTypeInformation.StorageType)).Returns((object)null);
+
+      commandMock.Setup(_ => _.CreateParameter()).Returns(dataParameterMock.Object).Verifiable();
+
+      dataParameterMock.SetupSet(_ => _.ParameterName = "@dummy").Verifiable();
+      dataParameterMock.SetupSet(_ => _.DbType = _storageTypeInformation.StorageDbType).Verifiable();
+      dataParameterMock.SetupSet(_ => _.Value = DBNull.Value).Verifiable();
+
+      var result = _dialect.CreateDataParameter(commandMock.Object, _storageTypeInformation, "@dummy","value");
+
+      commandMock.Verify();
+      dataParameterMock.Verify();
+
+      Assert.That(result, Is.SameAs(dataParameterMock.Object));
+    }
+
+    [Test]
+    public void CreateDataParameter_WithNullResult_AndParameterSize_SetsSize ()
+    {
+      var commandMock = new Mock<IDbCommand>(MockBehavior.Strict);
+      var dataParameterMock = new Mock<IDbDataParameter>(MockBehavior.Strict);
+
+      var storageTypeInformation = new StorageTypeInformation(
+          typeof(byte[]),
+          "test",
+          DbType.Binary,
+          false,
+          5,
+          typeof(byte[]),
+          _typeConverterStub.Object);
+
+      _typeConverterStub.Setup(_ => _.ConvertTo(null, null, "value", storageTypeInformation.StorageType)).Returns((object)null);
+
+      commandMock.Setup(_ => _.CreateParameter()).Returns(dataParameterMock.Object).Verifiable();
+
+      dataParameterMock.SetupSet(_ => _.ParameterName = "@dummy").Verifiable();
+      dataParameterMock.SetupSet(_ => _.DbType = storageTypeInformation.StorageDbType).Verifiable();
+      dataParameterMock.SetupSet(_ => _.Value = DBNull.Value).Verifiable();
+      dataParameterMock.SetupSet(_ => _.Size = storageTypeInformation.StorageTypeLength.Value).Verifiable();
+
+      var result = _dialect.CreateDataParameter(commandMock.Object, storageTypeInformation, "@dummy","value");
+
+      commandMock.Verify();
+      dataParameterMock.Verify();
+
+      Assert.That(result, Is.SameAs(dataParameterMock.Object));
+    }
+
+    [Test]
+    public void CreateDataParameter_WithNullInput ()
+    {
+      var commandMock = new Mock<IDbCommand>(MockBehavior.Strict);
+      var dataParameterMock = new Mock<IDbDataParameter>(MockBehavior.Strict);
+
+      _typeConverterStub.Setup(_ => _.ConvertTo(null, null, null, _storageTypeInformation.StorageType)).Returns("converted value");
```

Comment:
shouldn't this be DbNullValue? I believe right now, we're missing the DBNullValue test in this fixture.
We might also be missing DBNullValue with parameter size?